src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1083,6 +1084,51 @@ protected NodeModel(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPo
             RaisesModificationEvents = true;
         }
 
+        /// <summary>
+        /// Here we try to find the correct port names and tooltips.
+        /// ideally we'd use the runtime information to correctly update or localize
+        /// the port info, if we can't find it for any of the ports of the current node we fallback to the deserialized data
+        /// for all the ports.
+        /// Other strategies for port data lookup can be added here in the future.
+        /// </summary>
+        /// <param name="inPorts"></param>
+        /// <param name="outPorts"></param>
+        private void FindPortData(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts)
+        {
+            //try to get portData from attributes on the nodeModel type.
+            //to assign the correct tooltips.
+            //This might fail because some nodes don't use attributes to define 
+            //port data and instead add them using PortModel constructors.
+            var inportDatas = GetPortDataFromAttributes(PortType.Input).ToList();
+            var outportDatas = GetPortDataFromAttributes(PortType.Output).ToList();
+
+            // if the above attempt failed, for example because the node type does not have any port data attributes
+            // then give up for all ports.
+            if (inportDatas.Count() != inPorts.Count() || outportDatas.Count() != outPorts.Count())
+            {
+                return;
+            }
+
+            for (var i = 0; i < outPorts.Count() && i < outportDatas.Count; i++)
```

Comment:
At this point in time, won't the `outPorts `count and `outportDatas `count be the same?

Reply:
yes @aparajit-pratap you're right, now that we're bailing out if the counts don't match I should be able to simplify this.